sympy/functions/combinatorial/factorials.py

```diff
@@ -841,22 +850,23 @@ def _eval(self, n, k):
 
     @classmethod
     def eval(cls, n, k):
-        from sympy import gamma
         n, k = map(sympify, (n, k))
         d = n - k
         if d.is_zero or k.is_zero:
             return S.One
-
-        if (k - 1).is_zero:
-            return n
-        if k.is_integer:
-            if k.is_negative:
-                return S.Zero
-            if n.is_integer and n.is_nonnegative and d.is_negative:
-                return S.Zero
-            return cls._eval(n, k)
-
-        return gamma(n + 1)/(gamma(k + 1)*gamma(n - k + 1))
+        elif d.is_zero is False:
```

Comment:
What is the purpose of this?

Reply:
It's just for cases where d becomes None (cannot be determined) like (n, k) where n and k are integers, because for such cases we don't need to think of further check, directly return `binomial(n, k)`
Though removing it won't make a difference (I tried with few cases), I guess its just that we do not need any other kind of a check for such types.